lxd/images.go

```diff
@@ -3427,10 +3444,28 @@ func imageAliasesPost(d *Daemon, r *http.Request) response.Response {
 //	  "500":
 //	    $ref: "#/responses/InternalServerError"
 func imageAliasesGet(d *Daemon, r *http.Request) response.Response {
-	projectName := request.ProjectParam(r)
 	recursion := util.IsRecursionRequest(r)
 
 	s := d.State()
+
+	projectName := request.ProjectParam(r)
+	err := s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		projectHasImages, err := dbCluster.ProjectHasImages(ctx, tx.Tx(), projectName)
+		if err != nil {
+			return err
+		}
+
+		if !projectHasImages {
+			projectName = api.ProjectDefaultName
```

Comment:
Same question as above, but for `_, alias, err := tx.GetImageAlias(ctx, projectName, name, true)`

Reply:
And also `!userHasPermission(entity.ImageAliasURL(projectName, name))`

Reply:
There seems to be something odd going on here, because we're setting the effective project name into the context, but also overriding the requested project name and passing it into the `userHasPermission` check.

Reply:
I suppose im wondering why we are needing to pass the project name into `userHasPermission` at all, if the effective project name is now available in the context?

Reply:
Yeah I agree I need to keep the project query parameter separate as I have done in all the other cases. This is an oversight

Reply:
Fixed.